tools/topology/CMakeLists.txt

```diff
@@ -129,10 +129,12 @@ set(TPLGS
 	"sof-imx8-cs42888-mixer\;sof-imx8-cs42888-mixer"
 	"sof-imx8mp-wm8960-mixer\;sof-imx8mp-wm8960-mixer"
 	"sof-imx8mp-wm8960-kwd\;sof-imx8mp-wm8960-kwd"
-	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DPLATFORM=tgl\;-DAMP_SSP=1"
-	"sof-tgl-max98357a-rt5682\;sof-adl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DPLATFORM=adl\;-DAMP_SSP=2"
-	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682-pdm1\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DDMIC_DAI_LINK_16k_PDM=STEREO_PDM1\;\;-DPLATFORM=tgl\;-DAMP_SSP=1"
-	"sof-tgl-max98357a-rt5682\;sof-tgl-rt1011-rt5682\;-DCODEC=RT1011\;-DFMT=s24le\;\;-DPLATFORM=tgl\;-DAMP_SSP=1"
+	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DCH=2\;-DPLATFORM=tgl\;-DAMP_SSP=1"
+	"sof-tgl-max98357a-rt5682\;sof-adl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DCH=2\;-DPLATFORM=adl\;-DAMP_SSP=2"
+	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682\;-DCODEC=MAX98390\;-DFMT=s32le\;-DCH=4\;-DPLATFORM=adl\;-DAMP_SSP=1"
```

Comment:
why is the format s32le only here?

Reply:
@plbossart , Thanks,  I should add two here
"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682\;-DCODEC=MAX98390\;-DFMT=s32le\;-DCH=2\;-DPLATFORM=adl\;-DAMP_SSP=1"
"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682-4ch\;-DCODEC=MAX98390\;-DFMT=s32le\;-DCH=4\;-DPLATFORM=adl\;-DAMP_SSP=1"

Reply:
@plbossart . As per Maxim's tdm suggestion,  it's supported 2(slot 0/1) and 4(slot 0/1/2/3) speakers.
So we verified the 2 and 4 speakers board with same topology. if no concerns, I will update new one.

if (!strcmp(codec_dai->component->name, MAX_98390_DEV0_NAME)) {
                        /* DEV0 tdm slot configuration Right */
                        snd_soc_dai_set_tdm_slot(codec_dai, 0x01, 3, 4, 32);
                }
                if (!strcmp(codec_dai->component->name, MAX_98390_DEV1_NAME)) {
                        /* DEV1 tdm slot configuration Left */
                        snd_soc_dai_set_tdm_slot(codec_dai, 0x02, 3, 4, 32);
                }

                if (!strcmp(codec_dai->component->name, MAX_98390_DEV2_NAME)) {
                        /* DEVi2 tdm slot configuration Tweeter Right */
                        snd_soc_dai_set_tdm_slot(codec_dai, 0x04, 3, 4, 32);
                }
                if (!strcmp(codec_dai->component->name, MAX_98390_DEV3_NAME)) {
                        /* DEV3 tdm slot configuration Tweeter Left */
                        snd_soc_dai_set_tdm_slot(codec_dai, 0x08, 3, 4, 32);
                }